src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+                                                                                      isDataflowRule: false);
+
+        internal static DiagnosticDescriptor ObsoleteRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableAddedMessage,
+                                                                                      DiagnosticCategory.Interoperability,
+                                                                                      RuleLevel.BuildWarning,
+                                                                                      description: s_localizableObsoleteMessage,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+        internal static DiagnosticDescriptor RemovedRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableAddedMessage,
+                                                                                      DiagnosticCategory.Interoperability,
+                                                                                      RuleLevel.BuildWarning,
+                                                                                      description: s_localizableRemovedMessage,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(MinimumOsRule, ObsoleteRule, RemovedRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                var typeName = WellKnownTypeNames.SystemRuntimeInteropServicesRuntimeInformation;
+
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(typeName + "Helper", out var runtimeInformationType))
+                {
+                    runtimeInformationType = context.Compilation.GetOrCreateTypeByMetadataName(typeName);
+                }
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesOSPlatform, out var osPlatformType))
+                {
+                    return;
+                }
```

Comment:
`GetOrCreateTypeByMetadataName` and `TryGetOrCreateTypeByMetadataName` invoke the same functionality.

```suggestion
                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(typeName + "Helper", out var runtimeInformationType) ||
                    !context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesOSPlatform, out var osPlatformType))
                {
                    return;
                }
```

Reply:
`typeName + "Helper"` class just for unit testing, will be updated same as you suggested when real API can be consumed. I was need to load  runtimeInformationType again with `runtimeInformationType = context.Compilation.GetOrCreateTypeByMetadataName(typeName);` for testing the analyzer with runtime repo (there is no `typeName + "Helper"` class )

Reply:
Ah, my bad. I didn’t notice you were querying different type names. Hopefully it will be cleaned up before release.

Reply:
I have left TODO comment